test/v2/services/MovementsServiceSpec.scala

```diff
@@ -0,0 +1,558 @@
+      val result                                               = sut.createMovement(EORINumber("1"), MovementType.Arrival, invalidRequest)
+      val expected: Either[PersistenceError, MovementResponse] = Left(PersistenceError.UnexpectedError(Some(upstreamErrorResponse)))
+      whenReady(result.value) {
+        _ mustBe expected
+      }
+    }
+  }
+
+  "Getting a list of Arrival message IDs" - {
+
+    "when an arrival is found, should return a Right of the sequence of message IDs" in forAll(
+      arbitrary[EORINumber],
+      arbitrary[MovementId],
+      Gen.option(arbitrary[OffsetDateTime]),
+      Gen.listOfN(3, arbitrary[MessageSummary])
+    ) {
+      (eori, arrivalId, receivedSince, expected) =>
+        when(mockConnector.getMessages(EORINumber(any()), any(), MovementId(any()), any())(any(), any()))
+          .thenReturn(Future.successful(expected))
+
+        val result = sut.getMessages(eori, MovementType.Arrival, arrivalId, receivedSince)
+        whenReady(result.value) {
+          _ mustBe Right(expected)
+        }
+    }
+
+    "when an arrival is not found, should return a Left with ArrivalNotFound" in forAll(
+      arbitrary[EORINumber],
+      arbitrary[MovementId],
+      Gen.option(arbitrary[OffsetDateTime])
+    ) {
+      (eori, arrivalId, receivedSince) =>
+        when(mockConnector.getMessages(EORINumber(any()), any(), MovementId(any()), any())(any(), any()))
+          .thenReturn(Future.failed(UpstreamErrorResponse("not found", NOT_FOUND)))
+
+        val result = sut.getMessages(eori, MovementType.Arrival, arrivalId, receivedSince)
+        whenReady(result.value) {
+          _ mustBe Left(PersistenceError.MovementNotFound(arrivalId, MovementType.Arrival))
+        }
+    }
+
+    "on a failed submission, should return a Left with an UnexpectedError" in forAll(
+      arbitrary[EORINumber],
+      arbitrary[MovementId],
+      Gen.option(arbitrary[OffsetDateTime])
+    ) {
+      (eori, arrivalId, receivedSince) =>
+        val error = UpstreamErrorResponse("error", INTERNAL_SERVER_ERROR)
+        when(mockConnector.getMessages(EORINumber(any()), any(), MovementId(any()), any())(any(), any()))
+          .thenReturn(Future.failed(error))
+
+        val result = sut.getMessages(eori, MovementType.Arrival, arrivalId, receivedSince)
+        whenReady(result.value) {
+          _ mustBe Left(PersistenceError.UnexpectedError(thr = Some(error)))
+        }
+    }
+
+  }
+
+  "Getting a list of Arrivals (Movement) by EORI" - {
+
+    "when an arrival (movement) is found, should return a Right" in forAll(
+      Gen.listOfN(3, arbitrary[MovementSummary]),
+      Gen.option(arbitrary[OffsetDateTime]),
+      arbitrary[EORINumber]
+    ) {
+
+      (expected, updatedSinceMaybe, eori) =>
+        when(mockConnector.getMovements(eori, MovementType.Arrival, updatedSinceMaybe))
+          .thenReturn(Future.successful(expected))
+
+        val result = sut.getMovements(eori, MovementType.Arrival, updatedSinceMaybe)
```

Comment:
same with this file. Please try to combine arrival/departure test cases based on movementType. Ok with if not possible to combine. just wanted to highlight to reduce the lines of code.